src/app/services/tests/ActivesPartsService.spec.ts

```diff
@@ -1,84 +1,218 @@
+            // Given a partDao including active parts without our user
+            const user: string = 'creator';
+            await partDAO.set('joinerId', {
+                listMoves: [],
+                playerZero: 'someUser',
+                playerOne: 'someOtherUser',
+                result: 5,
+                turn: 0,
+                typeGame: 'P4',
+            });
 
             // when asking hasActivePart('our user')
-            const hasUserActiveParts: boolean = service.hasActivePart('creator');
+            const hasUserActiveParts: boolean = service.hasActivePart(user);
 
             // then we should learn that yes, he has some
             expect(hasUserActiveParts).toBeFalse();
+        }));
+    });
+    describe('getActivePartsObs', () => {
+        it('should notify about new parts', async() => {
+            // Given that we are observing active parts
+            let seenActiveParts: IPartId[] = [];
+            const activePartsSub: Subscription = service.getActivePartsObs()
+                .subscribe((activeParts: IPartId[]) => {
+                    seenActiveParts = activeParts;
+                });
+
+            // When a new part is added
+            const part: IPart = {
+                listMoves: [],
+                playerZero: 'creator',
+                playerOne: 'firstCandidate',
+                result: 5,
+                turn: 0,
+                typeGame: 'P4',
+            };
+            await partDAO.create(part);
+
+            // Then the new part has been observed
+            expect(seenActiveParts.length).toBe(1);
+            expect(seenActiveParts[0].doc).toEqual(part);
+
+            activePartsSub.unsubscribe();
         });
+        it('should notify about deleted parts', fakeAsync(async() => {
+            // Given that we are observing active parts, and there is already one part
+            const part: IPart = {
+                listMoves: [],
+                playerZero: 'creator',
+                playerOne: 'firstCandidate',
+                result: 5,
+                turn: 0,
+                typeGame: 'P4',
+            };
+            const partId: string = await partDAO.create(part);
+            let seenActiveParts: IPartId[] = [];
+            const activePartsSub: Subscription = service.getActivePartsObs()
+                .subscribe((activeParts: IPartId[]) => {
+                    seenActiveParts = activeParts;
+                });
+
+            // When an existing part is deleted
+            await partDAO.delete(partId);
+
+            // Then the deleted part is not considered as an active part
+            expect(seenActiveParts.length).toBe(0);
+
+            activePartsSub.unsubscribe();
+        }));
+        it('should preserve non-deleted upon a deletion', fakeAsync(async() => {
+            // Given that we are observing active parts, and there are already multiple parts
```

Comment:
Given a service observing* ?
Aussi, deux trois "then is" instead "then should" !

Reply:
updated